app/src/main/java/unipd/se18/ocrcamera/CameraActivity.java

```diff
@@ -156,6 +163,17 @@ public void onImage(CameraKitView cameraKitView, final byte[] photo) {
                     edit.putString("imagePath", filePath.trim());
                     edit.apply();
 
+                    // analyze the brightness   @author Balzan Pietro
+                    final Bitmap bitmapImage2 = bitmapImage;
+                    new Thread(new Runnable() {      // run the brightness recognition on a new thread to improve performance
```

Comment:
Is it really useful running these statements in a new Thread considered that the code of the BrightnessRecognition is in another java class and so in another Thread?

Reply:
Maybe i coded it wrong but when I tried just running the statements like you said, the performance wasn't improved: it still waited for the Recognition to end before launching the ResultActivity with the intent. It was like there wasn't a new automatic thread, so I made it manually.